### PR TITLE
chore: fix up learn backlinks

### PR DESCRIPTION
**Pull request by kendallstrautman**
## 🔗 Relevant links

- [Preview link](https://dev-portal-git-ksfixup-learn-backlinks-hashicorp.vercel.app) 🔎
- [Asana task](https://app.asana.com/0/1199634971449915/1203124800140554) 🎟️

## 🗒️ What

This PR fixes up backlinks to learn.hashicorp.com from proxied-dot-io data and other docs nav data files. 

## 🤷 Why

to help with seo juice post learn launch.

## Comments

I'm not sure about the 'downloads' pages, if those are even used anymore. I just did a find and replace.

### CODE REVIEW

File: src/views/tutorial-view/components/next-previous/helpers.ts [hunk 1]
```diff
@@ -60,8 +60,7 @@ export function getNextPrevious({
 	if (currentCollectionSection === 'well-architected-framework') {
 		finalLink = '/well-architected-framework'
 	} else if (currentCollectionSection === 'onboarding') {
-		/** @TODO - remove this once the tutorial library is released */
-		finalLink = 'https://learn.hashicorp.com/search'
+		finalLink = 'https://developer.hashicorp.com/tutorials/library'
```
Comment (kendallstrautman) [inline src/views/tutorial-view/components/next-previous/helpers.ts:63]:
```suggestion
		finalLink = '/tutorials/library'
```

File: src/components/_proxied-dot-io/consul/downloads-props/index.jsx [hunk 2]
```diff
@@ -8,27 +8,27 @@ export default function DownloadsProps(preMerchandisingSlot) {
 		getStartedLinks: [
 			{
 				label: 'CLI Quickstart',
-				href: 'https://learn.hashicorp.com/collections/consul/get-started-vms',
+				href: 'https://developer.hashicorp.com/consul/tutorials/get-started-vms',
```
Comment (kendallstrautman) [inline src/components/_proxied-dot-io/consul/downloads-props/index.jsx:11]: 💭 is this data still used? These should probably be relative links then

File: src/components/breadcrumb-bar/__fixtures__/terraform-plugin-sdkv2-nav-data.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
 	},
 	{
 		"title": "Tutorials: Custom Providers",
-		"href": "https://learn.hashicorp.com/collections/terraform/providers?utm_source=WEBSITE&utm_medium=WEB_IO&utm_offer=ARTICLE_PAGE&utm_content=DOCS"
+		"href": "https://developer.hashicorp.com/terraform/tutorials/providers?utm_offer=ARTICLE_PAGE&utm_medium=WEB_IO&utm_content=DOCS&utm_source=WEBSITE"
```
Comment (kendallstrautman) [inline src/components/breadcrumb-bar/__fixtures__/terraform-plugin-sdkv2-nav-data.json:11]: Oops looks like this is a test file, prob not a big deal tho

**Review by BRKalow — APPROVED** [review]
one minor suggestion, thanks for doing this!

**Review by ashleemboyer — APPROVED** [review]

File: src/components/_proxied-dot-io/consul/downloads-props/index.jsx [hunk 4]
```diff
@@ -8,27 +8,27 @@ export default function DownloadsProps(preMerchandisingSlot) {
...
 			{
 				label: 'View all Consul tutorials',
-				href: 'https://learn.hashicorp.com/consul',
+				href: 'https://developer.hashicorp.com/consul/tutorials',
 			},
 		],
 		tutorialLink: {
-			href: 'https://learn.hashicorp.com/consul',
+			href: 'https://developer.hashicorp.com/consul/tutorials',
 			label: 'View Tutorials at HashiCorp Learn',
```
Comment (kendallstrautman) [inline src/components/_proxied-dot-io/consul/downloads-props/index.jsx:32]:
```suggestion
			label: 'View Tutorials',
```

File: src/components/_proxied-dot-io/nomad/use-case-page/index.jsx [hunk 5]
```diff
@@ -11,7 +11,7 @@ export default function UseCaseLayout({ title, description, children }) {
 				links={[
 					{
 						text: 'Explore HashiCorp Learn',
```
Comment (kendallstrautman) [inline src/components/_proxied-dot-io/nomad/use-case-page/index.jsx:13]:
```suggestion
						text: 'Explore Tutorials',
```

File: src/components/_proxied-dot-io/nomad/use-case-page/index.jsx [hunk 6]
```diff
@@ -35,7 +35,7 @@ export default function UseCaseLayout({ title, description, children }) {
...
```
Comment (kendallstrautman) [inline src/components/_proxied-dot-io/nomad/use-case-page/index.jsx:36]:
```suggestion
						text: 'Explore Tutorials,
```

File: src/components/_proxied-dot-io/vault/footer/index.jsx [hunk 7]
```diff
@@ -11,7 +11,7 @@ export default function Footer({ openConsentManager }) {
 					<Link href="/api">
 						<a>API</a>
 					</Link>
-					<a href="https://learn.hashicorp.com/vault">Learn</a>
+					<a href="https://developer.hashicorp.com/vault/tutorials">Learn</a>
```
Comment (kendallstrautman) [inline src/components/_proxied-dot-io/vault/footer/index.jsx:14]:
```suggestion
					<a href="https://developer.hashicorp.com/vault/tutorials">Tutorials</a>
```

File: src/pages/_proxied-dot-io/boundary/downloads/index.jsx [hunk 8]
```diff
@@ -35,7 +35,7 @@ function DownloadsPage({ binaryReleases, desktopReleases }) {
 			product="boundary"
 			tutorialLink={{
 				label: 'View Tutorial on HashiCorp Learn',
```
Comment (kendallstrautman) [inline src/pages/_proxied-dot-io/boundary/downloads/index.jsx:37]:
```suggestion
				label: 'View Tutorials',
```

File: src/pages/_proxied-dot-io/nomad/downloads/enterprise.jsx [hunk 9]
```diff
@@ -37,7 +37,7 @@ function DownloadsPage({ product, releases, latestVersion }) {
 				/>
 			}
 			tutorialLink={{
-				href: 'https://learn.hashicorp.com/nomad',
+				href: 'https://developer.hashicorp.com/nomad/tutorials',
 				label: 'View Tutorials at HashiCorp Learn',
```
Comment (kendallstrautman) [inline src/pages/_proxied-dot-io/nomad/downloads/enterprise.jsx:41]:
```suggestion
				label: 'View Tutorials',
```

File: src/pages/_proxied-dot-io/nomad/downloads/index.jsx [hunk 10]
```diff
@@ -36,7 +36,7 @@ function DownloadsPage({ product, releases, latestVersion }) {
 				/>
 			}
 			tutorialLink={{
-				href: 'https://learn.hashicorp.com/nomad',
+				href: 'https://developer.hashicorp.com/nomad/tutorials',
 				label: 'View Tutorials at HashiCorp Learn',
```
Comment (kendallstrautman) [inline src/pages/_proxied-dot-io/nomad/downloads/index.jsx:40]:
```suggestion
				label: 'View Tutorials',
```

File: src/pages/_proxied-dot-io/vagrant/downloads/index.jsx [hunk 11]
```diff
@@ -37,7 +37,7 @@ function DownloadsPage({ product, releases, latestVersion }) {
 				/>
 			}
 			tutorialLink={{
-				href: 'https://learn.hashicorp.com/vagrant',
+				href: 'https://developer.hashicorp.com/vagrant/tutorials',
 				label: 'View Tutorials at HashiCorp Learn',
```
Comment (kendallstrautman) [inline src/pages/_proxied-dot-io/vagrant/downloads/index.jsx:41]:
```suggestion
				label: 'View Tutorials',
```

File: src/pages/_proxied-dot-io/vault/downloads/index.jsx [hunk 12]
```diff
@@ -41,7 +41,7 @@ function DownloadsPage({ product, releases, latestVersion }) {
 				/>
 			}
 			tutorialLink={{
-				href: 'https://learn.hashicorp.com/vault',
+				href: 'https://developer.hashicorp.com/vault/tutorials',
 				label: 'View Tutorials at HashiCorp Learn',
```
Comment (kendallstrautman) [inline src/pages/_proxied-dot-io/vault/downloads/index.jsx:45]:
```suggestion
				label: 'View Tutorials',
```

File: src/pages/_proxied-dot-io/waypoint/downloads/index.tsx [hunk 13]
```diff
@@ -41,7 +41,7 @@ function WaypointDownloadsPage({
 		/>
 	)
 	const tutorialLink = {
-		href: 'https://learn.hashicorp.com/waypoint',
+		href: 'https://developer.hashicorp.com/waypoint/tutorials',
 		label: 'View Tutorials at HashiCorp Learn',
```
Comment (kendallstrautman) [inline src/pages/_proxied-dot-io/waypoint/downloads/index.tsx:45]:
```suggestion
		label: 'View Tutorials',
```

**Pull request merged**